src/sds_data_model/_dataframe.py

```diff
@@ -10,6 +10,12 @@
 from rasterio.features import geometry_mask
 from xarray import DataArray
 
+from shapely.wkt import loads 
+from bng_indexer import calculate_bng_index, wkt_from_bng
+from pyspark.sql.types import ArrayType, FloatType, StringType
+from pyspark.sql.functions import col, explode, udf
```

Comment:
You don't use `StringType`,  `col` or `explode` either

```suggestion
from pyspark.sql.types import ArrayType, FloatType
from pyspark.sql.functions import udf